packages/vm/src/runBlock.ts

```diff
@@ -951,17 +957,143 @@ const DAOConfig = {
+
+    // Dequeue withdrawal requests
+
+    const queueHeadIndex = bytesToBigInt(
+      await vm.stateManager.getContractStorage(withdrawalsAddress, queueHeadSlot)
+    )
+    const queueTailIndex = bytesToBigInt(
+      await vm.stateManager.getContractStorage(withdrawalsAddress, queueTailSlot)
+    )
+    const numInQueue = queueTailIndex - queueHeadIndex
+    const numDequeued = Number(bigIntMin(numInQueue, maxWithdrawalsPerBlock))
+
+    for (let i = 0; i < numDequeued; i++) {
+      const queueStorageSlot = offset + (queueHeadIndex + BigInt(i)) * BIGINT_3
+      const sourceAddress = setLengthLeft(
+        await vm.stateManager.getContractStorage(
+          withdrawalsAddress,
+          setLengthLeft(bigIntToBytes(queueStorageSlot), 32)
+        ),
+        32
+      ).slice(12, 32)
+
+      const slot1Data = setLengthLeft(
+        await vm.stateManager.getContractStorage(
+          withdrawalsAddress,
+          setLengthLeft(bigIntToBytes(queueStorageSlot + BIGINT_1), 32)
+        ),
+        32
+      )
+      const slot2Data = setLengthLeft(
+        await vm.stateManager.getContractStorage(
+          withdrawalsAddress,
+          setLengthLeft(bigIntToBytes(queueStorageSlot + BIGINT_2), 32)
+        ),
+        32
+      )
+
+      const concatenatedData = concatBytes(slot1Data, slot2Data)
+      const validatorPubkey = concatenatedData.slice(0, 48)
+
+      const amount = unpadBytes(concatenatedData.slice(48, 56))
+
+      const RLPdBytes = RLP.encode([sourceAddress, validatorPubkey, amount])
+      const request = new ValidatorWithdrawalRequest(withdrawalsType, RLPdBytes)
+      requests.push(request)
+    }
+
+    const newQueueHeadIndex = queueHeadIndex + BigInt(numDequeued)
+
+    if (newQueueHeadIndex === queueTailIndex) {
+      await vm.stateManager.putContractStorage(withdrawalsAddress, queueHeadSlot, new Uint8Array())
+      await vm.stateManager.putContractStorage(withdrawalsAddress, queueTailSlot, new Uint8Array())
+    } else {
+      await vm.stateManager.putContractStorage(
+        withdrawalsAddress,
+        queueHeadSlot,
+        bigIntToBytes(newQueueHeadIndex)
+      )
+    }
+
+    // Update the excess withdrawal requests
+    const previousExcess = bytesToBigInt(
+      await vm.stateManager.getContractStorage(withdrawalsAddress, excessWithdrawalsSlot)
+    )
+    const count = bytesToBigInt(
+      await vm.stateManager.getContractStorage(withdrawalsAddress, withdrawalsCountSlot)
+    )
+
+    let newExcess = BIGINT_0
+    if (previousExcess + count > targetWithdrawalRequestsPerBlock) {
+      newExcess = previousExcess + count - targetWithdrawalRequestsPerBlock
+    }
+
+    await vm.stateManager.putContractStorage(
+      withdrawalsAddress,
+      withdrawalsCountSlot,
+      bigIntToBytes(newExcess)
+    )
+
+    // Reset the withdrawals count
+    await vm.stateManager.putContractStorage(
+      withdrawalsAddress,
+      withdrawalsCountSlot,
+      new Uint8Array()
+    )
+  }
```

Comment:
Can we move this whole code block to an internal helper method ˋ_accumulate7002WithdrawalsRequests()ˋ or something?

Reply:
Have addressed!